drivers/sensor/adi/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,904 @@
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Prepare data to send */
+	tx_buf[0] = addr;
+	tx_buf[1] = val;
+
+	/* Write register value */
+	const struct spi_buf tx_buf_arr = {
+		.buf = tx_buf,
+		.len = sizeof(tx_buf),
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf_arr,
+		.count = 1,
+	};
+
+	ret = spi_write_dt(&config->spi, &tx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int ad2s1210_set_resolution(const struct device *dev, enum ad2s1210_res resolution)
+{
+	const struct ad2s1210_config *config = dev->config;
+	struct ad2s1210_data *data = dev->data;
+	int ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	control &= ~(AD2S1210_CONTROL_RES_MASK);
+
+	switch (resolution) {
+	case AD2S1210_RES_10BIT:
+		break;
+	case AD2S1210_RES_12BIT:
+		control |= AD2S1210_CONTROL_RES1_MASK;
+		break;
+	case AD2S1210_RES_14BIT:
+		control |= AD2S1210_CONTROL_RES0_MASK;
+		break;
+	case AD2S1210_RES_16BIT:
+		control |= (AD2S1210_CONTROL_RES1_MASK | AD2S1210_CONTROL_RES0_MASK);
+		break;
+	default:
+		LOG_ERR("Invalid resolution: %d", resolution);
+		return -EINVAL;
+	}
+
+	ret = ad2s1210_reg_write(dev, AD2S1210_REG_CONTROL, control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!data->have_resolution_pins) {
+		data->resolution = resolution;
+		return 0;
+	}
+
+	/* Set RES0 pin */
+	ret = gpio_pin_set_dt(&config->resolution_gpios[AD2S1210_RES_PIN_RES0],
+			      !!(control & AD2S1210_CONTROL_RES0_MASK));
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Set RES1 pin */
+	ret = gpio_pin_set_dt(&config->resolution_gpios[AD2S1210_RES_PIN_RES1],
+			      !!(control & AD2S1210_CONTROL_RES1_MASK));
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->resolution = resolution;
+	return 0;
+}
+
+static int ad2s1210_set_hysteresis(const struct device *dev, bool enable)
+{
+	int ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	control &= ~AD2S1210_ENABLE_HYSTERESIS;
+	if (enable) {
+		control |= AD2S1210_ENABLE_HYSTERESIS;
+	}
+
+	return ad2s1210_reg_write(dev, AD2S1210_REG_CONTROL, control);
+}
+
+static __maybe_unused int
+ad2s1210_hysteresis_is_enabled(const struct device *dev) /* cppcheck-suppress unusedFunction */
+{
+	int ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (control & AD2S1210_ENABLE_HYSTERESIS) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static __maybe_unused int
+ad2s1210_reinit_excitation_frequency(const struct device *dev,
+				     uint16_t fexcit) /* cppcheck-suppress unusedFunction */
+{
+	const struct ad2s1210_config *config = dev->config;
+	int ret;
+	uint32_t fcw;
+
+	/* Calculate the frequency control word */
+	fcw = (fexcit * (1 << 15)) / config->clock_frequency;
+	if (fcw < AD2S1210_MIN_FCW || fcw > AD2S1210_MAX_FCW) {
+		return -ERANGE;
```

Comment:
`ERANGE` is usually used if a calculation overflows. `EINVAL` is used if the argument is invalid:
```suggestion
		return -EINVAL;
```

Reply:
Ok fixing it now.